src/main/java/com/blackduck/integration/detect/lifecycle/run/step/BinaryScanStepRunner.java

```diff
@@ -37,14 +45,45 @@ public Optional<String> runBinaryScan(
     )
         throws OperationException, IntegrationException {
         Optional<File> binaryScanFile = determineBinaryScanFileTarget(dockerTargetData, binaryTargets);
+        if (binaryScanFile.isPresent()) {            
+            // call BlackDuck to create a scanID and determine where to upload the file
+            ScanCreationResponse scanCreationResponse = operationRunner.initiateScan(projectNameVersion, binaryScanFile.get(), blackDuckRunData,
+                    "BINARY", gson);
+            
+            String scanId = scanCreationResponse.getScanId();
+            String uploadUrl = scanCreationResponse.getUploadUrl();
+            
+            if (StringUtils.isNotEmpty(uploadUrl)) {
+                // This is a SCASS capable server server and SCASS is enabled.
+                ScassScanStepRunner scassScanStepRunner = new ScassScanStepRunner(blackDuckRunData);
+                
+                scassScanStepRunner.runScassScan(binaryScanFile, scanCreationResponse);       
+            } else {
+                // This is a SCASS capable server server but SCASS is not enabled.
+                BdbaScanStepRunner bdbaScanStepRunner = new BdbaScanStepRunner(operationRunner);
+                
+                bdbaScanStepRunner.runBdbaScan(projectNameVersion, blackDuckRunData, binaryScanFile, scanId, "BINARY");
+            }
+            
+            return Optional.of(scanId);
+        } else {
+            return Optional.empty();
+        }
+    }
+    
+    public Optional<String> runLegacyMultipartBinaryScan(DockerTargetData dockerTargetData, NameVersion projectNameVersion,
```

Comment:
We should reconsider the use of the word "legacy". It confused me somewhat as I am getting more familiar with these changes.
Sometimes it means non-multipart and sometimes it means without SCASS abilities.